File: C++/binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> data;
    
    vector<int> postorderTraversal(TreeNode* root) {
        traverse(root);
        return data;
    }
    
    void traverse(TreeNode* root) {
        if (root) {
            traverse(root->left);            
            traverse(root->right);
            data.push_back(root->val);
        }
    }
};

File: Go/binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
	return post(root, []int{})
}

func post(node *TreeNode, values []int) []int {
	if node == nil {
		return values
	}
	values = post(node.Left, values)
	values = post(node.Right, values)
	values = append(values, node.Val)
	return values
}